### PR TITLE
refactor: handle reCAPTCHA settings and methods move to RAS

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Refactor required to handle the migration of reCAPTCHA settings to the Reader Activation wizard in https://github.com/Automattic/newspack-plugin/pull/1910.

### How to test the changes in this Pull Request:

This is technically a pure refactor. To test, check out this PR and https://github.com/Automattic/newspack-plugin/pull/1910 and follow the testing steps in https://github.com/Automattic/newspack-plugin/pull/1910 to confirm that reCAPTCHA still works as expected on streamlined donate blocks.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/blocks/donate/class-wp-rest-newspack-donate-controller.php [hunk 1]
```diff
@@ -99,60 +99,13 @@ public function register_routes() {
 	 * @return WP_REST_Response
 	 */
 	public function api_process_donation( $request ) {
-		// If reCaptcha is available, verify the user action.
-		$use_captcha = \Newspack\Stripe_Connection::can_use_captcha();
-		if ( $use_captcha ) {
-			$captcha_token = $request->get_param( 'captchaToken' );
-			if ( ! $captcha_token ) {
+		$captcha_token = $request->get_param( 'captchaToken' );
```
Comment (miguelpeixe) [inline src/blocks/donate/class-wp-rest-newspack-donate-controller.php:102]: If reCaptcha is enabled, it should fail if this param is empty

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: src/blocks/donate/class-wp-rest-newspack-donate-controller.php (hunk 1 above)
Comment (dkoo) [reply src/blocks/donate/class-wp-rest-newspack-donate-controller.php]: Updated in b50df46

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: src/blocks/donate/class-wp-rest-newspack-donate-controller.php [hunk 2]
```diff
@@ -99,60 +99,13 @@ public function register_routes() {
...
-
-			// If the reCaptcha verification score is below our threshold for valid user input.
-			if (
-				isset( $captcha_verify['score'] ) &&
-				Newspack_Blocks::DONATE_STREAMLINED_CAPTCHA_THRESHOLD > floatval( $captcha_verify['score'] )
-			) {
-				return rest_ensure_response(
-					[
-						'error' => __( 'User action failed captcha challenge.', 'newspack-blocks' ),
+						'error' => wp_strip_all_tags( $captcha_result->get_error_message() ),
```
[the remark this answers is not in the document]
Comment (dkoo) [reply src/blocks/donate/class-wp-rest-newspack-donate-controller.php]: Good point—updated in e9d2537 (and also added global namespaces to WP functions and classes).

**Review by miguelpeixe — APPROVED** [review]
🙌

**Pull request merged**